### PR TITLE
Move examples to a subfolder; add lex&parse example using lex library.

### PR DESCRIPTION
**Pull request by snakeru**
This example uses the lex.Lexer instead of a custom one. The reasons for the pull are two-fold:
1) There is no lexer+parser example for the languages with keywords yet. The most evolved example (calc) uses only operations and numbers, no keywords
2) Ability to mark the lexed candidate and then unread bytes on abort. I have lost the whole day before I realized what the problem was (the calc example is too primitive). Hopefully someone else will use my example.
Specifically I was trying to parse a C-like language that features divisions (single slash) and comments until end of the line (double slash).

### CODE REVIEW

File: examples/c-like/example.l [hunk 1]
```diff
@@ -0,0 +1,87 @@
+%{
+// Copyright (c) 2015 The golex Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+// This is an example program using golex run time library.
+package main
+
+import (
+  "bufio"
```
Comment (cznic) [inline examples/c-like/example.l]: Please gofmt.

File: examples/c-like/example.l [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+
+func newLexer(src io.Reader, dst io.Writer, fName string) *lexer {
+  file := token.NewFileSet().AddFile(fName, -1, 1<<31-1)
+  lx, err := lex.New(file, bufio.NewReader(src), lex.RuneClass(rune2Class))
+  if (err != nil) { panic(err) }
+  return &lexer{lx}
+}
+
+func (l *lexer) Token() string {
+  out := new(bytes.Buffer)
```
Comment (cznic) [inline examples/c-like/example.l]: Please use TokenBytes() instead.

**Review by cznic — CHANGES_REQUESTED** [review]
See inline notes. Also, please add yourself to the AUTHORS and/or CONTRIBUTORS file. Thank you.

File: examples/c-like/example.l (hunk 1 above)
Comment (cznic) [inline examples/c-like/example.l]: Err, this is not a .go file. Feel free to ignore the previous comment.

File: examples/c-like/example.l (hunk 2 above)
Comment (snakeru) [inline examples/c-like/example.l]: Done.

File: examples/c-like/example.l (hunk 1 above)
Comment (snakeru) [inline examples/c-like/example.l]: Ack.

**Comment by snakeru** [comment]
Just in case my 'Done' comments already propagated to you: please bare with me, it's the first time I use github for code reviews. I am currently researching what is the proper way to git commit --amend here.

**Comment by cznic** [comment]
> Just in case my 'Done' comments already propagated to you: please bare with me, it's the first time I use github for code reviews.

No problem and thanks for the PR!

**Comment by snakeru** [comment]
I amended my commit and did a force-push (hopefully that's the right way to do it - I was not able to find a clear instruction how to make changes and keep just a single commit).
Please take another look. The correct commit is bf73...

Thank you for the TokenBytes() hint! I was able to delete a few lines.

**Review by cznic — APPROVED** [review]
LGTM, thank you. Please merge to the master branch.

**Comment by snakeru** [comment]
Thanks! Done.

**Comment by cznic** [comment]
Thank you!

**Pull request merged**